Build error on OS X/macOS >= 10.11 and/or Xcode 8

When running macosx-build-dependencies.sh with the -f flag, it fails to build qt with the problem described in [this discussion](http://stackoverflow.com/questions/39492617/xcode-8-error-project-error-xcode-not-set-up-properly-even-though-the-licens).

The symptom error message is `Xcode not set up properly. You may need to confirm the license agreement by running /usr/bin/xcodebuild.`

The fix is, apparently, in ../libraries/src/qt-everywhere-opensource-src-5.7/qtbase/mkspecs/features/mac/default_pre.prf

```
    # Make sure Xcode is set up properly
+    isEmpty($$list($$system("/usr/bin/xcrun -find xcodebuild 2>/dev/null"))): \
-    isEmpty($$list($$system("/usr/bin/xcrun -find xcrun 2>/dev/null"))): \
        error("Xcode not set up properly. You may need to confirm the license agreement by running /usr/bin/xcodebuild.")
```

(manually copied/pasted/modified: this is not an automatic diff suitable for a patchfile)

I haven't actually implemented and/or tested the fix because I don't know the openscad build system, where to put patches, etc.

I just submitted a pull request.  Two files had to be patched, vs the singe file of QTBUG-56004

As I predicted, being able to make Qt got me just one step closer to being able to build OpenSCAD.  The build of QScintilla (the next dependency after Qt) fails.

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -c -pipe -stdlib=libc++ -std=c++11 -stdlib=libc++ -O2 -std=gnu++11 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.8 -fvisibility=hidden -fvisibility-inlines-hidden -w -fPIC -DQSCINTILLA_MAKE_DLL -DSCINTILLA_QT -DSCI_LEXER -DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_MACEXTRAS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I. -I. -I../include -I../lexlib -I../src -I../../../install/lib/QtPrintSupport.framework/Headers -I../../../install/lib/QtWidgets.framework/Headers -I../../../install/lib/QtMacExtras.framework/Headers -I../../../install/lib/QtGui.framework/Headers -I../../../install/lib/QtCore.framework/Headers -I. -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks/AGL.framework/Headers -I../../../install/mkspecs/macx-clang -F/Users/palmer/repo/notmine/3dprinting/libraries/install/lib -o qsciapis.o qsciapis.cpp
In file included from In file included from qsciscintilla.cpp:23:
./Qsci/qsciscintilla.h:2138:25: error: implicit instantiation of undefined template 'QPointer<QsciLexer>'
qsciapis.cpp:23:
In file included from ./Qsci/qsciapis.h:33:
In file included from ./Qsci/qsciabstractapis.h:33:
./Qsci/qsciscintilla.h:2138:25: error: implicit instantiation of undefined template 'QPointer<QsciLexer>'
    QPointer<QsciLexer> lex;    QPointer<QsciLexer> lex;
```

Next issue is glib:
```
configure: error:
*** GLib requires a 64 bit type. You might want to consider
*** using the GNU C compiler.
```

This turns out to be related to a new security feature in OS X 10.11, see e.g. http://apple.stackexchange.com/questions/215030/el-capitan-make-check-dyld-library-path

I don't have a fix for that yet. It looks like all shell scripts are considered protected binaries and thus are not allowed to be passed an DYLD_LIBRARY_PATH, meaning it cannot find the libraries we just built.

I can think of two workarounds:

1. Build our own version of bash to be able to run any script we want
2. Create all libraries using absolute install_name paths, and rewrite these to @rpath after we've built OpenSCAD binaries.

--- a/scripts/macosx-build-dependencies.sh
+++ b/scripts/macosx-build-dependencies.sh
@@ -43,7 +43,7 @@ PACKAGES=(
     "glew 1.13.0"
     "gettext 0.19.8"
     "libffi 3.2.1"
-    "glib2 2.46.1"
+    "glib2 2.50.1"
     "opencsg 1.4.2"
     "freetype 2.6.3"
     "ragel 6.9"
@@ -552,7 +552,7 @@ build_freetype()
   PKG_CONFIG_LIBDIR="$DEPLOYDOR/lib/pkgconfig" ./configure --prefix="$DEPLOYDIR" CFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN LDFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN $extra_config_flags
   make -j"$NUMCPU"
   make install
-  install_name_tool -id @rpath/libfreetype.dylib $DEPLOYDIR/lib/libfreetype.dylib
+  install_name_tool -id $DEPLOYDIR/lib/libfreetype.dylib $DEPLOYDIR/lib/libfreetype.dylib
 }
  
 check_libzip()
@@ -595,10 +595,10 @@ build_libxml2()
   fi
   tar xzf "libxml2-$version.tar.gz"
   cd "libxml2-$version"
-  ./configure --prefix="$DEPLOYDIR" --with-zlib=/usr -with-lzma --without-ftp --without-http --without-python CFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN LDFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN
+  ./configure --prefix="$DEPLOYDIR" --with-zlib=/usr --without-lzma --without-ftp --without-http --without-python CFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN LDFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN
   make -j$NUMCPU
   make install
-  install_name_tool -id @rpath/libxml2.dylib $DEPLOYDIR/lib/libxml2.dylib
+  install_name_tool -id $DEPLOYDIR/lib/libxml2.dylib $DEPLOYDIR/lib/libxml2.dylib
 }
 
 check_fontconfig()
@@ -667,10 +667,10 @@ build_gettext()
   tar xzf "gettext-$version.tar.gz"
   cd "gettext-$version"
 
-  ./configure --prefix="$DEPLOYDIR" CFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN LDFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN
+  ./configure --with-included-glib --prefix="$DEPLOYDIR" CFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN LDFLAGS=-mmacosx-version-min=$MAC_OSX_VERSION_MIN
   make -j$NUMCPU
   make install
-  install_name_tool -id @rpath/libintl.dylib $DEPLOYDIR/lib/libintl.dylib
+  install_name_tool -id $DEPLOYDIR/lib/libintl.dylib $DEPLOYDIR/lib/libintl.dylib
 }
 
 check_glib2()

